### PR TITLE
.

### PR DESCRIPTION
**Pull request by cfmnephrite**

### CODE REVIEW

File: test/ability_intimidate.c [hunk 1]
```diff
@@ -23,7 +23,7 @@ SINGLE_BATTLE_TEST("Intimidate (opponent) lowers player's attack after switch ou
         {
             ABILITY_POPUP(opponent, ABILITY_INTIMIDATE);
             ANIMATION(ANIM_TYPE_GENERAL, B_ANIM_STATS_CHANGE, player);
-            MESSAGE("Foe Arbok's Intimidate cuts Wobbuffet's attack!");
+            MESSAGE("Foe Arbok's ability cut \nWobbuffet's Attack!");
```
Comment (AsparagusEduardo) [inline test/ability_intimidate.c:26]: This is just ugly. The current test system allows for in-game like breaks to be interpreted as a space in the test name, which is much cleaner. This looks like a hacky solution.
I saw that your string is set as "cut " (with a space at the end), which is unneeded. You should be able to add a space to the string if needed.

File: include/constants/battle_move_effects.h [hunk 2]
```diff
@@ -197,7 +197,7 @@
 #define EFFECT_SNATCH                       193
 #define EFFECT_LOW_KICK                     194
 #define EFFECT_SECRET_POWER                 195
-#define EFFECT_RECOIL_33                    196
+#define EFFECT_UNUSED_196                   196
```
Comment (AsparagusEduardo) [inline include/constants/battle_move_effects.h:200]: Unused status should be removed and everything shifted.

Comment (cfmnephrite) [reply include/constants/battle_move_effects.h:200]: Yup, I'm just leaving that until the end so that I don't have to shift multiple times if I remove more

File: test/ability_intimidate.c (hunk 1 above)
Comment (cfmnephrite) [reply test/ability_intimidate.c:26]: Ohhh, it's the spaces causing the issues, not the newlines... OK, will adjust!

File: test/move_effect_defog.c [hunk 3]
```diff
@@ -135,8 +135,8 @@ DOUBLE_BATTLE_TEST("Defog lowers evasiveness by 1 and removes Stealth Rock and S
 {
     u16 move;
```
Comment (cfmnephrite) [inline test/move_effect_defog.c:137]: This test causes others to break... Not quite sure what to do about it

File: tools/mgba-rom-test-hydra/main.c [hunk 4]
```diff
@@ -101,7 +101,7 @@ static void handle_read(struct Runner *runner)
 add_to_results:
                     runner->results++;
                     soc += 2;
-                    fprintf(stdout, "%s: ", runner->test_name);
+                    fprintf(stdout, "[%d] %s: ", runner->pid, runner->test_name);
```
Comment (mrgriffin) [inline tools/mgba-rom-test-hydra/main.c:104]: I don't think you intended to commit this change. (It was used to work out which process was running which tests, but I think the changes to improve the "my tests fail when run together, but not when run alone" experience should be done on a different PR)

File: test/ability_defiant_competitive.c [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+        PLAYER(SPECIES_WIGGLYTUFF) { Ability(ABILITY_COMPETITIVE); };
+        OPPONENT(SPECIES_PRIMEAPE) { Ability(ABILITY_DEFIANT); };
+    } WHEN {
+        TURN { MOVE(player, MOVE_CHARM); }
+    } SCENE {
+        ABILITY_POPUP(opponent, ABILITY_DEFIANT);
+        ANIMATION(ANIM_TYPE_GENERAL, B_ANIM_STATS_CHANGE, opponent);
+        MESSAGE("Foe Primeape's Defiant sharply raised its Attack!");
+    } FINALLY {
+        EXPECT_EQ(opponent->statStages[STAT_ATK], DEFAULT_STAT_STAGE);
```
Comment (mrgriffin) [inline test/ability_defiant_competitive.c:18]: Somebody else brought this up on Discord, but it would be preferable (and consistent with the existing tests) if all these checks against `statStages` were instead done by observing a change in the damage dealt (this is normally done by `PARAMETRIZING` over whether the ability triggers or not).

Some other minor test style notes:
- `PARAMETRIZE {}` doesn't do anything, so you wouldn't have to include it.
- I don't think `ABILITY_COMPETITIVE` is being used for anything? If that's true, then it shouldn't be specified.
- We typically use `SPECIES_WOBBUFFET` for Pokémon where the species doesn't matter (i.e. Wigglytuff in this example).
- You don't need the `;`s after `}`s in `PLAYER` and `OPPONENT`.

**Review by mrgriffin — CHANGES_REQUESTED** [review]

File: tools/mgba-rom-test-hydra/main.c (hunk 4 above)
Comment (cfmnephrite) [reply tools/mgba-rom-test-hydra/main.c:104]: I left this in intentionally as it's quite useful and very trivial. If you really think I should remove it, I'll do so but I don't really see the need.

Comment (mrgriffin) [reply tools/mgba-rom-test-hydra/main.c:104]: I like printing something before the tests, but I think it'd be better if it was the test runner number (which is stable) rather than the PID (which isn't). If it was the runner number we could imagine being able to re-run the test suite in a mode that's like "run the tests which would be run by runner number X", and if we also inherited `make`'s `-j` parameter then we would be able to easily reproduce these sorts of errors on another developer's machine.

We definitely have a bad story for debugging these sorts of interferences between test cases, so if you want to take a look at doing a PR that improves the whole experience that would be great—if not, no worries I'll get to it eventually :)
